File: EECloud.PlayerIO/BigDB.cs
using EECloud.PlayerIO.Helpers;
using EECloud.PlayerIO.Messages;

namespace EECloud
{
    class BigDB
    {
        private: const string PlayerObjectsTableName;

        private: HttpChannel* _channel; //Readonly

        BigDB(HttpChannel* channel);//Constructor

        public: DatabaseObject* LoadMyPlayerObject();
    }
}

File: EECloud.PlayerIO/BigDB.cpp
#include "BigDB.hpp"

namespace EECloud
{
	BigDB::BigDB(HttpChannel* channel)
	{
		string PlayerObjectsTableName = "PlayerObjects";
		_channel = channel;
	}
	
	DatabaseObject BigDB::LoadMyPlayerObject()
	{
		LoadMyPlayerObjectOutput* loadMyPlayerObjectOutput = _channel->Request<NoArgsOrOutput*, LoadMyPlayerObjectOutput*, PlayerIOError*>(103, new NoArgsOrOutput());
		DatabaseObject output = loadMyPlayerObjectOutput->playerobject();
		output.set_table(PlayerObjectsTableName);
		delete loadMyPlayerObjectOutput;
		return output;
	}
}
